amaranth/vendor/_gatemate.py

```diff
@@ -0,0 +1,120 @@
+from abc import abstractmethod
+from amaranth import *
+from amaranth.build import *
+from amaranth.lib.cdc import ResetSynchronizer
+
+__all__ = ["GateMatePlatform"]
+
+class GateMatePlatform(TemplatedPlatform):
+    """
+    .. rubric:: GateMate FPGA toolchain
+
+    Required tools:
+        * ``yosys``
+        * ``p_r``
+
+    The environment is populated by setting the ``CC_TOOL`` environment variable to point to the toolchain directory.
+    """
+
+    device = property(abstractmethod(lambda: None))
+    package = property(abstractmethod(lambda: None))
+
+    toolchain = "GateMate"
+
+    required_tools = [
+        "yosys",
+        "p_r",
+    ]
+
+    file_templates = {
+        **TemplatedPlatform.build_script_templates,
+        "{{name}}.v": r"""
+            /* {{autogenerated}} */
+            {{emit_verilog()}}
+        """,
+        "{{name}}.debug.v": r"""
+            /* {{autogenerated}} */
+            {{emit_debug_verilog()}}
+        """,
+        "{{name}}.ccf": r"""
+            # {{autogenerated}}
+            {% for port_name, pin_name, attrs in platform.iter_port_constraints_bits() -%}
+                Net "{{port_name}}" Loc = "{{pin_name}}"
+                {%- set constraints = [
+                    "SCHMITT_TRIGGER", "PULLUP", "PULLDOWN", "KEEPER", "SLEW", "DRIVE", 
+                    "DELAY_OBF", "DELAY_IBF", "FF_IBF", "FF_OBF", "LVDS_BOOST", "LVDS_TERM"
+                ] -%}
```

Comment:
While this is a possible way to implement this functionality, in other platforms we don't filter attributes like this; instead, anything that is defined on the pin is dumped into the constraint file. Neither this solution nor our usual one offers any error checking, and our usual solution doesn't require updating Amaranth, releasing a backport, etc if the vendor adds a new parameter later.

Reply:
Removed the functionality, now it is like other platforms in Amaranth.
```suggestion
                | {{constraint}}={{value}}
                {%- endfor -%};
        {% endfor %}
        """,
```

Reply:
Thanks!